include/bluetooth/iso.h

```diff
@@ -180,30 +180,40 @@ struct bt_iso_chan_path {
 	uint8_t				cc[0];
 };
 
-/** ISO packet status flags */
+/** ISO packet status flag bits */
 enum {
 	/** The ISO packet is valid. */
-	BT_ISO_FLAGS_VALID,
+	BT_ISO_FLAGS_VALID = BIT(0),
+
 	/** @brief The ISO packet may possibly contain errors.
 	 *
 	 * May be caused by a failed CRC check or if missing a part of the SDU.
 	 */
-	BT_ISO_FLAGS_ERROR,
+	BT_ISO_FLAGS_ERROR = BIT(1),
+
 	/** The ISO packet was lost. */
-	BT_ISO_FLAGS_LOST
+	BT_ISO_FLAGS_LOST = BIT(2),
```

Comment:
These three are mutual exclusive - hence not sure it should be 3 individual bits, but rather a 2-bit value in a bitfield and a separate enum - like before.
For example:
```
...
uint8_t flags_status : 2
uint8_t flags_time_stamp : 1
...
```
Maybe you can even use the enum as the type - very old C-standards did not allow this, but I think more resent will - I believe it will generate an error if it does not.

Reply:
@Casper-Bonde-Bose I don't think mutually exclusive bits in a bitfield is a problem, but perhaps it should be documented? 

I did consider changing this to an enum as well. 

Regarding changing the field in the struct to 2 bitfield values, I think that is something we've generally avoided in the host, and have earlier been told not to do that, and to use macros instead. @jhedberg @Vudentz Any of you can chime in here why we avoid e.g. `uint8_t flags : 2;`? 

One alternative to this change is to leave the current flags as is, and add a `bool valid_ts` instead.

Reply:
I would prefer either bit-filed in the structure or separate members. Maybe the latter would be the "simple" solution that allows use of enum type for the status - without the need to look into if all compilers support this? 
Status flag should not be of type `uint_8` but the enum in both cases.

Reply:
Changing the type of the packet status can be done, but I'd rather do that in a separate PR if we decide to simply add a new field for the TS validity. 

I'd like to get feedback from one other person regarding this (@jhedberg @asbjornsabo ?), as this approach is not any different than e.g. the `struct k_work.flags` 

Basically the three options are

1) What this PR proposes:
```c
uint8_t flags; /* both status flags and a TS valid flag */
```

2) 
New bool/flag for TS valid
```c
uint8_t flags; /* unchanged */
bool ts_valid;
```

3)
Using the C-style bitfields 
```c
uint8_t flags_status : 2;
uint8_t flags_ts_valid : 1;
```

Reply:
I don't mind it being a single flags field, since it keeps the struct simpler, but I'll leave this up to you.

Reply:
IMHO for consistency in host APIs, single flags field as already in this PR is good for me.

Reply:
(I Approved before reading this discussion.)
I see Casper's point, and consider it as valid.  Having a bitfield often implies that one can set any combination of bits.
Alternative 2. is in that sense the cleanest.  But it does also seem a bit clunky, as one gets two variables for communicating the status.
Alternative 1 should be OK, if well documented.
(I do not see a need for 3.)